tree/ntuple/v7/test/ntuple_modelext.cxx

```diff
@@ -9,31 +9,43 @@ TEST(RNTuple, SchemaExtensionSimple)
       auto fieldPt = model->MakeField<float>("pt", 42.0);
 
       auto ntuple = RNTupleWriter::Recreate(std::move(model), "myNTuple", fileGuard.GetPath());
+      ntuple->Fill();
+      *fieldPt = 12.0;
+      ntuple->Fill();
+
       auto modelUpdater = ntuple->CreateModelUpdater();
       modelUpdater->BeginUpdate();
       std::array<double, 2> fieldArray = refArray;
       modelUpdater->AddField<std::array<double, 2>>("array", &fieldArray);
       modelUpdater->CommitUpdate();
 
       ntuple->Fill();
-      *fieldPt = 12.0;
+      *fieldPt = 1.0;
       fieldArray[1] = 1337.0;
       ntuple->Fill();
    }
 
    auto ntuple = RNTupleReader::Open("myNTuple", fileGuard.GetPath());
-   EXPECT_EQ(2U, ntuple->GetNEntries());
+   EXPECT_EQ(4U, ntuple->GetNEntries());
    EXPECT_EQ(4U, ntuple->GetDescriptor()->GetNFields());
+   EXPECT_EQ(0U, ntuple->GetModel()->GetField("pt")->GetGlobalRangeFirst());
+   EXPECT_EQ(2U, ntuple->GetModel()->GetField("array")->GetGlobalRangeFirst());
 
    auto pt = ntuple->GetView<float>("pt");
    auto array = ntuple->GetView<std::array<double, 2>>("array");
    EXPECT_EQ(42.0, pt(0));
    EXPECT_EQ(12.0, pt(1));
-   EXPECT_EQ(refArray, array(0));
-   EXPECT_EQ(1337.0, array(1)[1]);
+   EXPECT_EQ(12.0, pt(2));
+   EXPECT_EQ(1.0, pt(3));
+
+   // There is no on-disk data for this field before entry 2, i.e., entry range [0..1] yields zeroes
+   EXPECT_EQ((std::array<double, 2>{0.0, 0.0}), array(0));
+   EXPECT_EQ((std::array<double, 2>{0.0, 0.0}), array(1));
+   EXPECT_EQ(refArray, array(2));
+   EXPECT_EQ(1337.0, array(3)[1]);
 }
 
-TEST(RNTuple, SchemaExtensionInvalidUse)
+TEST(RNTuple, ModelExtensionInvalidUse)
```

Comment:
Is the name of the test still reflecting the content?

Reply:
Indeed, it tests that an invalid use of the interface throws (e.g., trying to fill an entry where there's an open transaction).